compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -836,7 +846,13 @@ impl<'a, 'mir, 'tcx, M: Machine<'mir, 'tcx>> std::fmt::Debug for DumpAllocs<'a,
                             )?;
                         }
                         Some(GlobalAlloc::Function(func)) => {
-                            write!(fmt, " (fn: {})", func)?;
+                            write!(fmt, " (fn: {func})")?;
+                        }
+                        Some(GlobalAlloc::Vtable(ty, Some(trait_ref))) => {
+                            write!(fmt, " (vtable: impl {trait_ref} for {ty})")?;
+                        }
+                        Some(GlobalAlloc::Vtable(ty, None)) => {
+                            write!(fmt, " (vtable: impl ? for {ty})")?;
```

Comment:
Maybe document somewhere that the `None` case means we only get a [trivial vtable (drop, size, align) because it's just auto traits and no actual traits](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/list/struct.List.html#method.principal).

Reply:
🤷 either syntax works for me.

Wrt the `None` case, yes, adding some comments on the `Vtable` variant seems good. The output should probably just be `vtable for auto traits for {ty}` or something like this?

also I just noticed sometimes you use `VTable` and other times `Vtable`, should probably use consistent capitalization.

Reply:
I always used `Vtable` but some existing code might be using `VTable`. Other existing code [uses `Vtable`](https://github.com/rust-lang/rust/blob/4f0a64736b94a84da4272867366b8679b585e603/compiler/rustc_typeck/src/check/cast.rs#L72) like me, so... which part do you like me to be consistent with?

Reply:
The lowercase version

Reply:
FWIW the stdlib has `RawWakerVTable` which is table. So I guess uppercase it is...